js/main.js

```diff
@@ -0,0 +1,43 @@
+var $photoUrl = document.querySelector('#photo-url');
+$photoUrl.addEventListener('input', function () {
+
+  var $photoEntry = document.querySelector('#photoEntry');
+  $photoEntry.setAttribute('src', $photoUrl.value);
+
+});
+
+var $entryForm = document.querySelector('#journal-entry');
+$entryForm.addEventListener('submit', handleSubmit);
+
+function handleSubmit(event) {
+  event.preventDefault();
+  var titleInput = $entryForm.elements.title.value;
+  var photoUrlInput = $entryForm.elements.photoUrl.value;
+  var notesInput = $entryForm.elements.notes.value;
+
+  var journalEntry = {
+    title: titleInput,
+    photoUrl: photoUrlInput,
+    notes: notesInput,
+    nextEntryId: data.nextEntryId
+  };
+
+  journalEntry.nextEntryId = data.nextEntryId++;
+
+  data.entries.unshift(journalEntry);
+
+  var $photoEntry = document.querySelector('#photoEntry');
+  $photoEntry.setAttribute('src', 'images/placeholder-image-square.jpg');
+
+  $entryForm.reset();
+
+  // LOCAL STORAGE
+
+  window.addEventListener('beforeunload', function () {
+    var journalEntriesData = JSON.stringify(data);
+    localStorage.setItem('journal entry', journalEntriesData);
+  });
```

Comment:
this functionality should be done in the data.js, and you also need to load the data when the app starts.